modules/budget/main.tf

```diff
@@ -17,18 +17,23 @@
 locals {
   project_name     = length(var.projects) == 0 ? "All Projects" : var.projects[0]
   display_name     = var.display_name == null ? "Budget For ${local.project_name}" : var.display_name
-  all_updates_rule = var.alert_pubsub_topic == null && var.monitoring_notification_channels == null ? [] : ["1"]
+  all_updates_rule = var.alert_pubsub_topic == null && length(var.monitoring_notification_channels) == 0 ? [] : ["1"]
 
   projects = length(var.projects) == 0 ? null : [
-    for id in var.projects :
-    "projects/${id}"
+    for project in data.google_project.project :
```

Comment:
What I think you're going to need to do is:

Keep the `for_each` the same (iterating over the project_id).

Inside the resource itself, look up the number from the data source.

Reply:
I was able to repro using the integration tests. Not all tests are working (I don't think the errors are related to my changes) but the billing test seems ok now.

Switching from `for_each` to `count` seems to solve the issue.

Reply:
Thanks, looks like that worked.